packages/waku/src/router/define-router.ts

```diff
@@ -110,14 +110,16 @@ export function unstable_defineRouter(
   ) => {
     const pathConfig = await pathConfigPromise;
     const path2moduleIds: Record<string, string[]> = {};
-    for (const { path: pathSpec } of pathConfig) {
+    for (const { path: pathSpec, isStatic } of pathConfig) {
       if (pathSpec.some(({ type }) => type !== 'literal')) {
         continue;
       }
       const pathname = '/' + pathSpec.map(({ name }) => name).join('/');
       const input = getInputString(pathname);
-      const moduleIds = await unstable_collectClientModules(input);
-      path2moduleIds[pathname] = moduleIds;
+      if (isStatic) {
+        const moduleIds = await unstable_collectClientModules(input);
+        path2moduleIds[pathname] = moduleIds;
+      }
```

Comment:
Hmm, I think we want to use `path2moduleIds` for dynamic pages too (even though, it's suboptimal).

If `unstable_collectClientModules` is only the place of the problem, it doesn't actually leak the information because it's thrown away.
That said, I wish there would be an alternative way to collect client modules without running the code. Can we statically analyze the code?

Reply:
Oh, yes we can. I read the React-server source code, we could collect client info during vite build

Reply:
my current idea is to add a hook on the resolve module function and give id -> moduleId[] Map in the analysis plugin

Reply:
that sounds nice. we may need some research, so this would come at v0.21 or later.